### ISSUE TITLE
Support for latest Gradle

### ISSUE THREAD

**Issue by pratikpparikh** [jonnyzzz/gradle-java9c#1]
Dear author,

In my testing, it seems that your plugin is not current with the 4.x version of Gradle.  Could you please provide the update.  When use it produces the following stack trace in debugging.

```
08:10:45.483 [DEBUG] [org.gradle.internal.progress.DefaultBuildOperationExecutor] Build operation 'Resolve files of :engage:runtimeClasspath' started
08:10:45.483 [DEBUG] [org.gradle.internal.progress.DefaultBuildOperationExecutor] Completing Build operation 'Resolve files of :engage:runtimeClasspath'
08:10:45.484 [DEBUG] [org.jonnyzzz.gradle.java9c.Resolver] Failed to complete a resolution try: Resolving configuration 'implementation' directly is not allowed
java.lang.IllegalStateException: Resolving configuration 'implementation' directly is not allowed
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.assertResolvingAllowed(DefaultConfiguration.java:924)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.access$1400(DefaultConfiguration.java:116)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration$ConfigurationArtifactCollection.<init>(DefaultConfiguration.java:1212)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration$ConfigurationArtifactCollection.<init>(DefaultConfiguration.java:1208)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration$ConfigurationResolvableDependencies.getArtifacts(DefaultConfiguration.java:1069)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration$ConfigurationResolvableDependencies_Decorated.getArtifacts(Unknown Source)
        at org.jonnyzzz.gradle.java9c.ResolveConfigurationIncomingDependencies$resolve$1$2.invoke(resolver.kt:165)
        at org.jonnyzzz.gradle.java9c.ResolveConfigurationIncomingDependencies$resolve$1$2.invoke(resolver.kt:147)
        at kotlin.sequences.FlatteningSequence$iterator$1.ensureItemIterator(Sequences.kt:252)
        at kotlin.sequences.FlatteningSequence$iterator$1.hasNext(Sequences.kt:240)
        at kotlin.sequences.FilteringSequence$iterator$1.calcNext(Sequences.kt:107)
        at kotlin.sequences.FilteringSequence$iterator$1.hasNext(Sequences.kt:132)
        at org.jonnyzzz.gradle.java9c.ResolveConfigurationIncomingDependencies.resolve(resolver.kt:185)
        at org.jonnyzzz.gradle.java9c.Resolver$fileCollectionResolver$3.invoke(resolver.kt:93)
        at org.jonnyzzz.gradle.java9c.Resolver$fileCollectionResolver$3.invoke(resolver.kt:76)
        at org.jonnyzzz.gradle.java9c.Resolver.tryAndHandle(resolver.kt:112)
        at org.jonnyzzz.gradle.java9c.Resolver.fileCollectionResolver(resolver.kt:91)
        at org.jonnyzzz.gradle.java9c.ScannerKt.scanPackages(scanner.kt:43)
        at org.jonnyzzz.gradle.java9c.ScanClasspathTask.execute java9c task(java9c.kt:72)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:780)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:747)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:121)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:60)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:97)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:87)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:626)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:581)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
        at java.base/java.lang.Thread.run(Thread.java:844)
08:10:45.493 [DEBUG] [org.gradle.internal.progress.DefaultBuildOperationExecutor] Completing Build operation 'Execute execute java9c task for :engage:java9c_main'
08:10:45.493 [DEBUG] [org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter] Removed task artifact state for {} from context.
08:10:45.493 [DEBUG] [org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter] Finished executing task ':engage:java9c_main'
08:10:45.494 [DEBUG] [org.gradle.internal.progress.DefaultBuildOperationExecutor] Completing Build operation 'Task :engage:java9c_main'
08:10:45.494 [null] [org.gradle.internal.progress.DefaultBuildOperationExecutor]
08:10:45.494 [DEBUG] [org.gradle.internal.progress.DefaultBuildOperationExecutor] Build operation 'Task :engage:java9c_main' completed
08:10:45.494 [INFO] [org.gradle.execution.taskgraph.DefaultTaskPlanExecutor] :engage:java9c_main (Thread[Task worker for ':',5,main]) completed. Took 0.071 secs.
08:10:45.494 [DEBUG] [org.gradle.internal.work.DefaultWorkerLeaseService] Worker lease root.1.9 completed (1 worker(s) in use)
08:10:45.494 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Task worker for ':': released lock on root.1.9
08:10:45.495 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Task worker for ':': released lock on :
0
```

Regards,
Pratik Parikh

**Comment by jonnyzzz** [comment]
What is full version of grade where you have the issue?

You may share a minimal example to help me reproducing the problem

Does it failing or do you only see the error in debug logs?

**Comment by pratikpparikh** [comment]
Hi Jonny,

Thank you, Jonny, for a quick response.

**Below is the full gradle version**
```
C:\Users\prati\git\nubes>gradle -version
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/C:/opt/development/build/gradle/lib/groovy-all-2.4.12.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

------------------------------------------------------------
Gradle 4.4.1
------------------------------------------------------------

Build time:   2017-12-20 15:45:23 UTC
Revision:     10ed9dc355dc39f6307cc98fbd8cea314bdd381c

Groovy:       2.4.12
Ant:          Apache Ant(TM) version 1.9.9 compiled on February 2 2017
JVM:          9.0.1 (Oracle Corporation 9.0.1+11)
OS:           Windows 10 10.0 amd64
```

**I only see the following error, not the analysis output.**

`08:10:45.499 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
08:10:45.499 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * What went wrong:
08:10:45.499 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Execution failed for task ':engage:java9c_main'.
08:10:45.499 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > begin 0, end -1, length 17`

**You can use the following open source project [java9-vertx](https://github.com/java9-modularity/java9-vertx) and add  id "org.jonnyzzz.java9c" version "0.2.1" to plugins list in easytext.web** then execute the gradle java9c and obtain the following output.  I am using JDK 9.0.1
```
C:\Users\prati\git\java9-vertx\easytext.web>gradle java9c
Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':easytext.web:java9c_main'.
> begin 0, end -1, length 17

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 19s
14 actionable tasks: 2 executed, 12 up-to-date
```

**Comment by jonnyzzz** [comment]
Thank you for the info. I fixed the problem and released `0.2.2` builds. Please give it a try
